resource/js/components/PageEditor/ContextBasedNewLineHandlerExecutor.js

```diff
@@ -0,0 +1,29 @@
+import * as codemirror from 'codemirror';
+
+import markdownListHelper from './MarkdownListHelper';
+import markdownTableHelper from './MarkdownTableHelper';
+
+/**
+ * Selector for one new line handler
+ */
+class ContextBasedNewLineHandlerExecutor {
```

Comment:
You should implement such executer as a subclass of `BasicInterceptor`.
Reference `InterceptorManager` and `DetachCodeBlockInterceptor`.

Reply:
Thank you, I'll referred.